lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1169,6 +1169,33 @@ swift::matchWitness(WitnessChecker::RequirementEnvironmentCache &reqEnvCache,
   // Match a type in the requirement to a type in the witness.
   auto matchTypes = [&](Type reqType,
                         Type witnessType) -> llvm::Optional<RequirementMatch> {
+    // `swift_attr` attributes in the type context were ignored before,
+    // which means that we need to maintain status quo to avoid breaking
+    // witness matching by stripping everything concurrency related from
+    // inner types.
+    if (req->isObjC()) {
```

Comment:
Can we limit this stripping behavior to Swift 5 without strict concurrency?

Reply:
Done!